klpbuild/klplib/ibs.py

```diff
@@ -593,11 +594,24 @@ def create_lp_package(self, i, cs):
 
         logging.info(f"({i}/{self.total}) {cs.name()} done")
 
-    def log(self, cs, arch):
+    def log(self, arch):
+        cs_list = filter_codestreams(self.lp_filter, get_codestreams_dict())
+
+        if not cs_list:
+            logging.error("log: No codestreams found for filter %s", self.lp_filter)
+            sys.exit(1)
+
+        if len(cs_list) > 1:
+            cs_names = [cs.name() for cs in cs_list]
+            logging.error("Filter '%s' returned %d entries (%s), while expecting just one. Aborting. ",
+                          self.lp_filter, len(cs_list), " ".join(cs_names))
+            sys.exit(1)
+
+        cs = Codestream.from_cs(cs_list[0].name())
```

Comment:
Hi, I just realized that using `from_cs` here is not needed.

You are retrieving the name from a `Codestream` object (`cs_list[0].name()`), and then from this name you are instantiating a new one (`Codestream.from_cs()`), right?

Could you just do:

```suggestion
        cs = cs_list[0]
```
perhaps?

If that's the case, we can totally drop the function `drop_cs` and further simplify the codebase :)